library/platform_util.c

```diff
@@ -125,6 +125,26 @@ void mbedtls_platform_zeroize(void *buf, size_t len)
         SecureZeroMemory(buf, len);
 #else
         memset_func(buf, 0, len);
+#endif
+
+#if defined(__GNUC__)
+        /* For clang and recent gcc, pretend that we have some assembly that reads the
+         * zero'd memory as an additional protection against being optimised away. */
+#if defined(__clang__) || (__GNUC__ >= 10)
```

Comment:
Searching online, it seems that "invalid argument to gimple call" comes from a bug in some versions of GCC 4.x and 5.x. So `__GNUC__ >= 6` seems safe.

I tested locally on x86_64 and x86_32 under Ubuntu 16.04/18.04 and got the error with gcc 4.7.4 but not 5.4.0, 6.5.0, 7.5.0 or 8.4.0. This doesn't prove much since it could depend on the architecture and the optimization level.

I'm ok with staying with `>= 10` since we have no reports that earlier (or even current) compiler versions make an undesirable optimization here, so this is future-proofing, and does not need to apply to compiler versions that aren't actively developed any more.